spark/src/test/scala/org/apache/spark/sql/delta/DeltaInsertIntoColumnOrderSuite.scala

```diff
@@ -68,8 +68,9 @@ class DeltaInsertIntoColumnOrderSuite extends DeltaInsertIntoTest {
       overwriteWhere = "a" -> 1,
       insertData = TestData("a long, c int, b int", Seq("""{ "a": 1, "c": 4, "b": 5 }""")),
       expectedResult = ExpectedResult.Success(expectedAnswer),
-      // Dataframe insert by name don't support implicit cast, see negative test below.
-      includeInserts = inserts -- insertsByName.intersect(insertsDataframe)
+      // Exclude dataframe inserts by name (except streaming) which don't support implicit cast.
+      // See negative test below.
+      includeInserts = inserts -- (insertsByName.intersect(insertsDataframe) - StreamingInsert)
```

Comment:
Nit: This is getting a little bit complex. It's hard for me to understand for me which cases are actually covered here. It's okay to have some duplication in tests.

Reply:
Minor improvement: I introduced a variable for "inserts that don't support implicit casting" to make it a bit easier to reason about.